### PR TITLE
🎉 Source Oracle: Added support for connection via SSH tunnels

### LINKED ISSUE #5968: Add support & test cases for Oracle Source via SSH tunnels
## Tell us about the problem you're trying to solve
We'd like to add support for SSH tunnels for the existing Oracle source. This will allow connecting to the source from the different VPC. 

## Describe the solution you’d like
There are few main steps to accomplish this requirement:

1. Create a DB in AWS through terraform or the UI. We already have a blueprint for how to create a database on AWS in [terraform](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/infrastructure/ssh_tunnel/module/database.tf). You would need only to create a database without a public IP in the subnet which already exists, the same one the Postgres DB is currently linked to. Please talk to Oleksandra for getting access to the AWS dev environment where you can spin up a database. If you go this route, you can re-use many of the parameters (tunnel host, tunnel key, etc..) in the lastpass secrets `Postgres Source ssh key creds` and `destination postgres test creds with sshkey`. If Terraform is slowing you down too much feel free to create one in the UI.
2. Wrap existing Oracle source with `SshWrappedSource` and check that additional SSH configuration properties are present on UI
3. Cover this change with an appropriate number of `unit/integration/acceptance` tests

## Describe the alternative you’ve considered or used
Depending on whether you are adding a source or a destination the impl is slightly different. See the [instructions for how to add SSH support to a connector](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/ssh/readme.md)

[Example PR of adding SSH support to the Postgres source](https://github.com/airbytehq/airbyte/pull/5742)

**Note that this PR has way more logic in them than what you'll need to implement for a single connector because they add common helpers and shared libraries and docs**

### PR DESCRIPTION
**Pull request by etsybaev**
## What
The goal of this PR is to add SSH only for the `Oracle Source`. It does not try to deal with how it can be reused for other sources or destinations.
Took the concept of My PR for Oracle source

## How
Handles injecting the SSH tunnel in the check, discover, read methods of `Oracle source` directly. This makes it relatively straightforward to inject without worrying about touching other jdbc dbs.

Tested with  AWS EC2 bastion and RDS Oracle + test normal connection.

Got 2 new options to connect via SSH (with key or pass)
Selection_042
Selection_043

Here is what was in the source DB
Selection_044
Selection_046

Here is the result of data migration via SSH
Selection_045

Also checked that direct connection to DB still works:
Selection_047

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-oracle/src/test-integration/java/io/airbyte/integrations/source/oracle/AbstractSshOracleSourceAcceptanceTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import io.airbyte.protocol.models.JsonSchemaPrimitive;
+import io.airbyte.protocol.models.SyncMode;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+public abstract class AbstractSshOracleSourceAcceptanceTest extends SourceAcceptanceTest {
+
+  // private static final String STREAM_NAME = "test.id_and_name";
```
Comment (alexandr-shegeda) [inline airbyte-integrations/connectors/source-oracle/src/test-integration/java/io/airbyte/integrations/source/oracle/AbstractSshOracleSourceAcceptanceTest.java:48]: please, remove commented code

**Review by alexandr-shegeda — APPROVED** [review]

File: docs/integrations/sources/oracle.md [hunk 2]
```diff
@@ -100,6 +100,23 @@ Your database user should now be ready for use with Airbyte.
 
 Case sensitive. Defaults to the upper-cased user if empty. If the user does not have access to the configured schemas, no tables will be discovered.
 
+## Connection to Oracle via an SSH Tunnel
```
Comment (sherifnada) [inline docs/integrations/sources/oracle.md:103]:
```suggestion
## Connection via an SSH Tunnel
```

File: docs/integrations/sources/oracle.md [hunk 3]
```diff
@@ -100,6 +100,23 @@ Your database user should now be ready for use with Airbyte.
...
+3. `SSH Tunnel Jump Server Host` refers to the intermediate (bastion) server that Airbyte will connect to. This should be a hostname or an IP Address.
+4. `SSH Connection Port` is the port on the bastion server with which to make the SSH connection. The default port for SSH connections is `22`, so unless you have explicitly changed something, go with the default.
+5. `SSH Login Username` is the username that Airbyte should use when connection to the bastion server. This is NOT the Oracle username.
+6. If you are using `Password Authentication`, then `SSH Login Username` should be set to the password of the User from the previous step. If you are using `SSH Key Authentication` leave this blank. Again, this is not the Oracle password, but the password for the OS-user that Airbyte is using to perform commands on the bastion.
+7. If you are using `SSH Key Authentication`, then `SSH Private Key` should be set to the RSA Private Key that you are using to create the SSH connection. This should be the full contents of the key file starting with `-----BEGIN RSA PRIVATE KEY-----` and ending with `-----END RSA PRIVATE KEY-----`.
+
 
 ## Changelog
```
Comment (sherifnada) [inline docs/integrations/sources/oracle.md:121]: we should update changelog too

File: airbyte-integrations/connectors/source-oracle/src/test-integration/java/io/airbyte/integrations/source/oracle/SshPasswordOracleSourceAcceptanceTest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+package io.airbyte.integrations.source.oracle;
+
+import java.nio.file.Path;
+
+public class SshPasswordOracleSourceAcceptanceTest extends AbstractSshOracleSourceAcceptanceTest {
+
+  @Override
+  public Path getConfigFilePath() {
+    return Path.of("secrets/ssh-pwd-config.json");
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-oracle/src/test-integration/java/io/airbyte/integrations/source/oracle/SshPasswordOracleSourceAcceptanceTest.java:33]: shouldn't this secret be injected in `ci_credentials.sh`?  where is it being inserted in CI?

**Review by sherifnada — APPROVED** [review]

File: docs/integrations/sources/oracle.md (hunk 3 above)
Comment (etsybaev) [reply docs/integrations/sources/oracle.md:121]: Hi Sherif.
Sure, I was going to to update versions, changelog and secrets before publishing

File: airbyte-integrations/connectors/source-oracle/src/test-integration/java/io/airbyte/integrations/source/oracle/SshPasswordOracleSourceAcceptanceTest.java (hunk 4 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-oracle/src/test-integration/java/io/airbyte/integrations/source/oracle/SshPasswordOracleSourceAcceptanceTest.java:33]: Hi Sherif.
Sure, I was going to update versions, changelog, and secrets before publishing. Secrets themselves are already added to LastPass and GitHub like ORACLE_SSH_KEY_TEST_CREDS и ORACLE_SSH_PWD_TEST_CREDS.
For json file name, I just follow the pattern of how it was done in other connectors

**Comment by sherifnada** [comment]
btw, great work on the screenshots! makes it very easy to see what's happening

File: docs/integrations/sources/oracle.md (hunk 2 above)
Comment (etsybaev) [reply docs/integrations/sources/oracle.md]: updated, thanks

**Pull request merged**